test/e2e/test_job_submissions.py

```diff
@@ -1084,27 +1130,29 @@ def test_worker_streams_logs_to_cloudwatch(
         )
 
         job.wait_until_complete(client=deadline_client)
-        logs_client: boto3.client = boto3.client(
-            "logs",
-            config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
-        )
 
         # Retrieve job output and verify the echo is printed
-        job_logs = job.get_logs(deadline_client=deadline_client, logs_client=logs_client)
-        full_log: str = "\n".join(
-            [le.message for _, log_events in job_logs.logs.items() for le in log_events]
-        )
 
-        assert (
-            full_log.count("HelloWorld") == 1
-        ), "Expected number of HelloWorld statements not found in job logs."
+        job.assert_single_task_log_contains(
+            deadline_client=deadline_client,
+            logs_client=boto3.client(
+                "logs",
+                config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
+            ),
+            expected_pattern=r"HelloWorld",
+        )
 
         # Retrieve worker logs and verify that it's not empty
         worker_log_group_name: str = (
             f"/aws/deadline/{deadline_resources.farm.id}/{deadline_resources.fleet.id}"
         )
         worker_id = session_worker.worker_id
 
+        logs_client: boto3.client = boto3.client(
+            "logs",
+            config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
+        )
```

Comment:
nit: looks like we already create a logs boto3 client on lines 1138-1142. We could share the same logs client.

Reply:
Updated